scripts/solve_network.py

```diff
@@ -182,6 +189,9 @@ def run_lopf(n, allow_warning_status=False, fix_ext_lines=False):
             add_lv_constraint(n)
             add_lc_constraint(n)
 
+        if extra_functionality is not None:
+            extra_functionality(n, *extra_functionality_args)
```

Comment:
Is there a specific reason for passing `extra_functionality_args`? The pypsa `extra_functionality` takes per default only `n` and `snapshots` . Perhaps keeping those would make sense and then there can be more keyword arguments `extra_functionality_args`, so 
```
extra_functionality(n, snapshots,  *extra_functionality_args)
```